integrations/slack/websockets/socketmode.go

```diff
@@ -0,0 +1,88 @@
+package websockets
+
+import (
+	"fmt"
+
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/socketmode"
+	"go.uber.org/zap"
+
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+// handler is an autokitteh webhook which implements [http.Handler]
+// to receive and dispatch asynchronous event notifications.
+type handler struct {
+	logger        *zap.Logger
+	secrets       sdkservices.Secrets
+	dispatcher    sdkservices.Dispatcher
+	integrationID sdktypes.IntegrationID
+	scope         string
+}
+
+func NewHandler(l *zap.Logger, sec sdkservices.Secrets, d sdkservices.Dispatcher, scope string, id sdktypes.IntegrationID) handler {
+	return handler{
+		logger:        l,
+		secrets:       sec,
+		dispatcher:    d,
+		scope:         scope,
+		integrationID: id,
+	}
+}
+
+// Key = Slack app ID (to ensure one WebSocket per app).
+var webSocketClients = make(map[string]*socketmode.Client)
+
+func (h handler) OpenSocketModeConnection(appID, botToken, appToken string) {
+	// No need to open multiple connections for the same app - yet.
+	// See: https://docs.autokitteh.com/tutorials/new_connections/slack.
+	if _, ok := webSocketClients[appID]; ok {
+		return
+	}
+
+	api := slack.New(botToken, slack.OptionAppLevelToken(appToken))
+	webSocketClients[appID] = socketmode.New(api)
+
+	go func() {
+		for {
+			smh := socketmode.NewSocketmodeHandler(webSocketClients[appID])
+			smh.HandleDefault(h.socketModeHandler)
+			err := smh.RunEventLoop()
+			if err != nil {
+				h.logger.Error("Slack Socket Mode (re)connection error", zap.Error(err))
```

Comment:
?

Reply:
Bug fixed - `continue`, not `return` if `RunEventLoop()` returns (i.e. reconnection error).

Also reversed condition to use an implicit `continue`, as requested.

Verified - goroutine is indeed needed.

Not done: can't merge `err := ...` and `if` now that the usage of `err` is outside the condition block